cachi2/core/package_managers/rpm/main.py

```diff
@@ -462,3 +480,50 @@ def _generate_repofiles(
 
             with open(repo_file_path, "w") as f:
                 repofile.write(f)
+
+
+def _get_ssl_context(options: Optional[dict] = None) -> ssl.SSLContext:
+
+    ssl_ctx = ssl.create_default_context()
+
+    if options is None:
+        log.debug("returning default ssl context.")
+        return ssl_ctx
+    else:
+        client_cert = options["ssl"]["client_cert"]
+        client_key = options["ssl"]["client_key"]
+        ca_bundle = options["ssl"]["ca_bundle"]
+        ssl_verify = options["ssl"]["ssl_verify"]
+
+    if client_cert is None and client_key is None:
+        log.debug("No client certificates will be used.")
+    elif path.isfile(path=client_cert) and path.isfile(path=client_key):
+        ssl_ctx.load_cert_chain(client_cert, client_key)
+        log.info("Using client certificate auth.")
+    else:
+        raise SSLContextError(
+            f"Specified client credentials files:\n \
+              ['{client_cert}']\n \
+              ['{client_key}']\n \
+              could not be opened.",
+            solution=("Ensure the files exist and that you have permission to the files."),
+        )
+
+    if ca_bundle is not None:
```

Comment:
Seems like we're checking multiple files if the path exists repeatedly, how about:
```
for ssl_file in [ca_bundle, client_cert, client_key]:
  if ssl_file is not None  and not Path(ssl_file).isfile():
    raise SSLContextError(...)
```

Then later in the code you'd only need
```
if ca_bundle is not None:
  ssl_ctx.load_verify_locations(ca_bundle)
```

Reply:
definitelye should not be using an image from my quay repo, but I don't have the necessary permissions to put it anywhere else and I assume we need to get these reviews cleared before someone from the cachi2 team is going to push it for me.  And, I had to put it somewhere...

There is also test data in git (rpm lockfile) that needs to be moved to the repo where the other rpm test data is.

Reply:
> When you use the official nginx docker image you don't need to worry about ^this.

But there's quite a lot of setup that needs to be done on top of the official image, right? It might be easier to just move the image setup to a repo in the [cachito-testing](https://github.com/cachito-testing/) org, we could try using Github's registry to store it, and a small Github Actions pipeline to build it.

Reply:
> But there's quite a lot of setup that needs to be done on top of the official image, right?

A lot of setup justifying a custom build image hosted in cachito-testing org's registry making it harder to actually understand what's happening? I don't think so. If I enumerate what's needed:

- curl a single RPM package
    ^this is trivial
- call `createrepo_c` on the temporary directory we downloaded the RPM from previous step to
   ^also straight forward, we already have a hard dependency on `createrepo_c` so we can safely assume it's installed
- mount volumes with certs, nginx, and DNF repo to the container
  ^the only difference to a custom built image is the last volume, the rest we have to do anyway
- check the aliveness of the container
  ^we have to do this either way
- run the integration test

In summary, unless I have overlooked some important bit (could happen easily) then I don't think my proposal adds any overhead over what's being proposed at all. Quite the contrary, the setup is fully transparent, relying on a small, official docker image designated to one thing and one thing only with any modifications being achieved with a set of data volumes mounted on startup which we'd unarguably do either way.

Reply:
If it is as simple as you say, and if we can avoid rebuilding the image on every CI run, then I think it's a good way to go. 

Otherwise, offloading it to another repo would avoid the need for rebuilds, and to have all these test files in the Cachi2 repo. The sources for the test would still be one click away, so I don't think it would make it hard to understand (we can leave a reference to the test repo in a comment). The part I'm not sure about is the Github registry, as I have never used it before.

Reply:
> If it is as simple as you say, and if we can avoid rebuilding the image on every CI run, then I think it's a good way to go.

For a comparison, have a look at this commit on my branch where I did just that without the need for a custom container image: https://github.com/containerbuildsystem/cachi2/commit/a276ab62db81025dd579feba96953e0526543c58